core/src/main/java/io/questdb/cairo/sql/RecordCursorFactory.java

```diff
@@ -90,6 +90,26 @@ default boolean fragmentedSymbolTables() {
         return false;
     }
 
+    default String getBaseColumnName(int idx) {
+        return getBaseFactory().getMetadata().getColumnName(idx);
+    }
+
+    /**
+     * Method is necessary for cases where row cursor uses index from table reader while record cursor can reorder columns (e.g. DataFrameRecordCursorFactory)
+     *
+     * @param idx idx of column
+     * @return name of base column (no remapping)
+     */
+    default String getBaseColumnNameNoRemap(int idx) {
+        return getBaseColumnName(idx);
+    }
+
+    default RecordCursorFactory getBaseFactory() {
```

Comment:
could you review implementations? There are a lot of duplicate and inconsistent methods

Reply:
done